### PR TITLE
grsecurity: update to linux-4.4

### PR DESCRIPTION
**Pull request by tg-x**
###### Things done:
- [ ] Tested via `nix.useChroot`.
- [x] Built on platform(s): x86_64.
- [ ] Tested compilation of all pkgs that depend on this change.
- [ ] Tested execution of binary products.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).
###### Extra

Fixes broken grsecurity due to kernel version change.

(FIXED: I was going to submit this, but by the time I created this pull request @fpletz updated the kernel version to 4.4.3, for which there's no grsecurity patch out yet.)

### CODE REVIEW

**Comment by fpletz** [comment]
Sorry. We should decouple grsecurity kernels from the mainline kernels. We have this problem with grsecurity for quite some time.

**Comment by copumpkin** [comment]
Yeah please decouple!

**Comment by tg-x** [comment]
alright, decoupled it

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -10407,6 +10407,24 @@ let
...
+
+  linux_grsecurity_4_4 = callPackage ../os-specific/linux/kernel/linux-grsecurity-4.4.nix {
+    kernelPatches = [ kernelPatches.bridge_stp_helper ]
+      ++ lib.optionals ((platform.kernelArch or null) == "mips")
+      [ kernelPatches.mips_fpureg_emu
+        kernelPatches.mips_fpu_sigill
+        kernelPatches.mips_ext3_n32
+      ];
+  };
+
```
Comment (NeQuissimus) [inline pkgs/top-level/all-packages.nix]: Could we have a `linux_grsecurity_latest` similar to what the standard kernel has?

Comment (tg-x) [inline pkgs/top-level/all-packages.nix]: well, can name it _stable & _testing instead of the version numbers

Comment (NeQuissimus) [inline pkgs/top-level/all-packages.nix]: No, what I mean is we have linuxPackages_latest that is simply an "alias" for whatever the latest is.
4_4 would be the specific 4.4 kernel, while latest switches to 4.5 as soon as 4.5 becomes stable.
That way configuring to latest simply requires a rebuild without a configuration change to update.

Comment (tg-x) [inline pkgs/top-level/all-packages.nix]: These are only used to refer to the appropriate base kernel for grsec internally in patches.nix,
the kernelPackages configuration so far has been set with:

```
kernelPackages = pkgs.linux_grsec_testing_server;
kernelPackages = pkgs.linux_grsec_testing_desktop;
```

So these don't include any version information, but would be indeed good to add it.

Comment (tg-x) [inline pkgs/top-level/all-packages.nix]: OK, fixed the versioning, too.

**Comment by fpletz** [comment]
Thanks! Merged, updated to 4.4.4 and also pushed to the 16.03 branch because @domenkozar set the milestone. :beers:

**Pull request merged**